### PR TITLE
Trigger preparation "immediately" when mounting whilst running

### PR DESCRIPTION
**Pull request by foolswood**
It would have got there in the end, but it was waiting for UINT32_MAX ms before the preparation loop would be rerun.

p.s. with this you can get noise from the haskell

### CODE REVIEW

File: langext/src/countdown.c [hunk 1]
```diff
@@ -49,6 +49,10 @@ int rage_countdown_add(rage_Countdown * c, int delta) {
     return val;
 }
 
-void rage_countdown_unblock_wait(rage_Countdown * c) {
-    sem_post(&c->sig);
+int rage_countdown_unblock_wait(rage_Countdown * c) {
+    int val = atomic_exchange(&c->counter, 0);
```
Comment (ch3pjw) [inline langext/src/countdown.c:53]: This is a little mysterious for me - googling turns up a lot of C++! Remind me to ask you about it on Monday.

**Review by ch3pjw — COMMENTED** [review]
Looks definitely like you changes some things to do with the semaphores/timing/preparation! This is another one of those ones that leaves my scratching my head though ;-)
I was half expecting some test change too to cover the changes in behaviour?

**Comment by foolswood** [comment]
The reordered srt tests actually catch a surprising amount of this (especially since it's pretty much all races). The unblock_wait method had been refactored out of use, but I'd left it around because I had a feeling it'd come in handy. It is testable, but as with all things countdowny what you have to write is a thready pain.

File: langext/src/countdown.c (hunk 1 above)
Comment (foolswood) [reply langext/src/countdown.c:53]: It puts 0 in the counter and gives you a copy of the previous value atomically.

**Pull request merged**